usecase/syncstrategy/window.go

```diff
@@ -46,42 +45,33 @@ func (window *Window) Sync(
 	})
 	logger.Debug("spawning goroutines for sync block workers")
 
+	workersWaitGroup := sync.WaitGroup{}
+	workersWaitGroup.Add(int(endHeight - beginHeight + 1))
+	commandWindowMutex := sync.Mutex{}
+
+	commandWindow := newUnsafeCommandWindow(beginHeight, endHeight)
+
 	for height := beginHeight; height <= endHeight; height += 1 {
 		go func(height int64) {
 			commands, err := worker(height)
+			result := workResult{height, commands, nil}
+
 			if err != nil {
-				workResultCh <- workResult{height, nil, err}
+				logger.Errorf("received error from sync block worker #%d: %v", height, err)
+				result = workResult{height, nil, err}
 			}
 
```

Comment:
I think the error is swallowed silently here.

Reply:
I missed it, thanks for pointing out. Added error handler.